client/src/pages/messaging/index.js

```diff
@@ -205,27 +210,28 @@ const Chat = () => {
       getSetUsers(channelID);
     });
 
-    socket.on('chat-message', (msg) => {
+    socket.on('chat-message', (msg, callback) => {
       try {
         const box = strToTypedArr(msg.message.box);
         const nonce = strToTypedArr(msg.message.nonce);
-
         const { msg: _msg } = decryptMsg({
           box,
           nonce,
           mySecretKey: myKeyRef.current.secretKey,
           alicePublicKey: publicKeyRef.current
         });
-
         setMessages((prevMsg) =>
           prevMsg.concat({
             image: msg.image,
             body: _msg,
-            sender: msg.sender
+            sender: msg.sender,
+            id: msg.id
           })
         );
+        callback({ ok: true });
```

Comment:
So, here you know that the client received the message. Emit an event that the message is delivered.